pkg/autodiscovery/providers/env.go

```diff
@@ -0,0 +1,79 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2018 Datadog, Inc.
+
+package providers
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/DataDog/datadog-agent/pkg/autodiscovery/integration"
+	"github.com/DataDog/datadog-agent/pkg/config"
+)
+
+const name = "dd_logs_config_custom_config"
+
+var collectors = []func(*EnvVarProvider) ([]integration.Config, error){
+	(*EnvVarProvider).collectLogsConfigs,
+	(*EnvVarProvider).collectMetricsConfigs,
+}
+
+// EnvVarProvider implements implements the ConfigProvider interface
+// It should be called once at the start of the agent.
+type EnvVarProvider struct{}
+
+// NewEnvVarProvider creates a EnvVarProvider searching for
+// configurations in env variables.
+func NewEnvVarProvider() *EnvVarProvider {
+	return &EnvVarProvider{}
+}
+
+// Collect gets the value of env variables
+// and generate an integrationConfig out of it.
+func (e *EnvVarProvider) Collect() ([]integration.Config, error) {
+	var integrationConfigs []integration.Config
+	var errors []string
+	var err error
+	for _, collector := range collectors {
+		collectorConfigs, err := collector(e)
```

Comment:
`collector(e)` looks weird, I'd rather like `collector()`

Reply:
Cannot reply to https://github.com/DataDog/datadog-agent/pull/2565#discussion_r232303358
but this is automatically done by `gofmt`, probably because the array syntax is `[]<type>{}` without space between the type and the braces